### PR TITLE
No more cdts6 qt6 and Do not pass -DQT_REQUIRE_HOST_PATH_CHECK=OFF + libbbost 1.86

### PR DESCRIPTION
**Pull request by hmaarrfk**
Closes #291  
Closes #286
Closes #279 

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -21,7 +21,8 @@ source:
     folder: opengl32sw                                                                                            # [win64]
 
 build:
-  number: 4
+  number: 5
+  skip: true  # [not linux]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:25]:
```suggestion
```

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
Not sure why aarch is failing with some sha mismatch.... twice....

But linux64's "configuration" as stated by Qt seems to be at parity with the main branch. The one difference seems to be:

* Main: `EGLFS EGLDevice ..... no`
* this branch: `EGLFS EGLDevice .... yes`

in
```
2024-08-25T01:26:20.7221148Z QPA backends:
2024-08-25T01:26:20.7221390Z   DirectFB ............................... no
2024-08-25T01:26:20.7221680Z   EGLFS .................................. yes
2024-08-25T01:26:20.7221921Z   EGLFS details:
2024-08-25T01:26:20.7222184Z     EGLFS OpenWFD ........................ no
2024-08-25T01:26:20.7222448Z     EGLFS i.Mx6 .......................... no
2024-08-25T01:26:20.7222737Z     EGLFS i.Mx6 Wayland .................. no
2024-08-25T01:26:20.7223019Z     EGLFS RCAR ........................... no
2024-08-25T01:26:20.7223295Z     EGLFS EGLDevice ...................... yes
```

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml [hunk 2]
```diff
@@ -73,51 +71,13 @@ requirements:
...
-    - xorg-libxrandr                    # [(build_platform != target_platform) and linux]
-    - xorg-libxcomposite                # [(build_platform != target_platform) and linux]
-    - xorg-libxcursor                   # [(build_platform != target_platform) and linux]
-    - xorg-libsm                        # [(build_platform != target_platform) and linux]
-    - xorg-libice                       # [(build_platform != target_platform) and linux]
-    - xorg-libxxf86vm                   # [(build_platform != target_platform) and linux]
-    - xorg-libxtst                      # [(build_platform != target_platform) and linux]
-    - xorg-xf86vidmodeproto             # [(build_platform != target_platform) and linux]
+    # circular dependency... but it helps when cross compiling
+    - qt6-main                          # [build_platform != target_platform]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:75]:
```suggestion
    - qt6-main   {{ version }}          # [build_platform != target_platform]
```

**Comment by jschueller** [comment]
do you think this would work with a new version eg 6.8 ?

**Comment by hmaarrfk** [comment]
we would have to "build it twice" the first time, only the native architectures would build, then we wait 2 hours, then restart the failed builds.

and if we are lucky, it works an additional PR.

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
I think something in the double build is corrupting the arm environment sadly....

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Review by jschueller — APPROVED** [review]
lets go on then

**Comment by hmaarrfk** [comment]
Thanks @jschueller

**Comment by hmaarrfk** [comment]
Needs: https://github.com/qt/qttools/commit/687fc1601863ae7a67897bc3590b33bd3bdcc3bc

**Pull request merged**